mpqp/execution/result.py

```diff
@@ -161,7 +161,10 @@ def __init__(
                     )
 
     def __str__(self):
-        return f"State: {self.bin_str}, Index: {self.index}, Count: {self.count}, Probability: {self.probability}"
+        return (
+            f"State: {self.bin_str}, Index: {self.index}, Count: {self.count}"
+            + f", Probability: {np.round(self.probability, 5) if self.probability is not None else None}"
```

Comment:
We create the probability or count in the `Result` and not in `Sample`, so the `Sample` can have `self.probability` None. We want to create the probability in the result because we need all the samples.
```py
self._probabilities = np.array(counts, dtype=float) / self.shots
for sample in self._samples:
   sample.probability = self._probabilities[sample.index]
```

Reply:
Indeed. See discussion above, but I still think that we never have `self.probability` to `None`, even in your new version of the code, right?
If the user only put the counts, then the probas will be computed by us.
If the user inputs both counts and probas, then the probas will be filled directly from there.
Correct me if I am wrong.